### ISSUE TITLE
[Refactor] Consider removing the `opensearch-dashboards` module from this repo

### ISSUE THREAD

**Issue by cwperks** [opensearch-project/OpenSearch#14500]
### Please describe the end goal of this project

There is a module in core called [opensearch-dashboards](https://github.com/opensearch-project/OpenSearch/tree/main/modules/opensearch-dashboards) that appears to be unused.

This plugin is a single file and all it does is register a few RestHandlers that wrap common operations around indices. On the surface, it looks like the purpose of this plugin is to facilitate requests from OpenSearch-Dashboards to allow dashboards to write to the saved objects (.kibana or .opensearch_dashboards*) index, but its not actually used by dashboards.

When running a cluster with security, dashboards must be configured with `opensearch.username` and `opensearch.password` which is used to write and read from the saved objects index or indices if using multi-tenancy. In the demo configuration, this is the `kibanaserver` user which is mapped to the [kibana_server](https://github.com/opensearch-project/security/blob/main/src/main/resources/static_config/static_roles.yml#L79-L118) role which grants it permission on those indices.

This is the list of [RestHandlers](https://github.com/opensearch-project/security/blob/main/src/main/resources/static_config/static_roles.yml#L79-L118) that it registers and it prefixes all of their existing routes with `_opensearch_dashboards/`

i.e. The create index API that this plugins registers is `PUT /_opensearch_dashboards/{idx}` instead of `PUT/{idx}`

Judging by the wrapper in this class it looks like the plugin is trying to provide additional protections to the [saved objects indices](https://github.com/opensearch-project/OpenSearch/blob/main/modules/opensearch-dashboards/src/main/java/org/opensearch/dashboards/OpenSearchDashboardsModulePlugin.java#L167-L170). 

If the module is unused then is there any purpose in keeping the code in core? 

### Supporting References

Related to work in https://github.com/opensearch-project/security/issues/4439 that is analyzing system indices across all default modules and plugins

### Issues

Related to https://github.com/opensearch-project/security/issues/4439

### Related component

Plugins

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/vikasvb90/) [3](https://github.com/srikanthpadakanti)]
@cwperks Thanks for creating this issue, I've updated the title to reflect that we'd like comments on this issue before making a decision

**Comment by andrross** [comment]
@reta Do you know the purpose of this module?

A [simple search](https://github.com/search?q=org%3Aopensearch-project+%22_opensearch_dashboards%22&type=code) for the `_opensearch_dashboards` prefix picks up a bunch of false positives, but digging through the results I haven't found anything that appears to actually use that prefix.

**Comment by reta** [comment]
> @reta Do you know the purpose of this module?

I actually don't to be fair, but it says 'Plugin exposing APIs for OpenSearch Dashboards system indices'

**Comment by nibix** [comment]
My five cents:

I am pretty sure that the purpose of this plugin is to mark the internal indices known as `.kibana`, `.kibana_*` or believed to be known as `.opensearch_dashboards`, etc. as system indices, i.e., to restrict access to these.

Before the fork, there were unfinished efforts to enhance system index protection. One intermediate goal was to allow system index access only if the client (in this case the internal Dashboards server), would (insecurely, just by a REST header) identify as a certain product. 

I have not checked it, but I actually believe that this intermediate goal is alread reached when for fork occurred. However, this module is inoperable due to the mismatch between the believed index name and the one that is still being used.

**Comment by cwperks** [comment]
@nibix that's correct and looks like the work is incompleted as dashboards does direct index access today.

FYI I've been thinking about this more and I think that [Resource Sharing and Access Control](https://github.com/opensearch-project/security/issues/4500) that security is introducing will ultimately provide a path here for using this plugin. A couple of problems to tackle with regards to OSD:

1. **Make `.kibana*` (or `.opensearch_dashboards*`) system indices** - These indices today are not formally system indices and requests against these may be authorized as normal indices.
2. **Introduce finer grained access controls** - i.e. introduce notion of ownership to `Saved Objects` like dashboards and visualizations. The creator of a dashboards can then specify sharing with other users in the system.
     - Today the access control is done at the tenant level. A user either has write access to the tenant (write access to all saved objects) or read access which prevents them from making any changes. 

With Resource Sharing and Access Control, an assumption is made that a _Sharable Resource_ is simply a doc with metadata stored in a system index and the `.kibana` indices would qualify.
